lib/assets/javascripts/turbograft/page.coffee

```diff
@@ -25,3 +25,47 @@ Page.refresh = (options = {}, callback) ->
 
 Page.open = ->
   window.open(arguments...)
+
+# Providing hooks for objects to set up destructors:
+onReplaceCallbacks = []
+
+# e.g., Page.onReplace(node, unbindListenersFnc)
+# unbindListenersFnc will be called if the node in question is partially replaced
+# or if a full replace occurs.  It will be called only once
+Page.onReplace = (node, callback) ->
+  throw new Error("Page.onReplace: Node and callback must both be specified") if !node || !callback
+  throw new Error("Page.onReplace: Callback must be a function") if !isFunction(callback)
+  onReplaceCallbacks.push({node, callback})
+
+# option C from http://jsperf.com/alternative-isfunction-implementations
+isFunction = (object) ->
+  !!(object && object.constructor && object.call && object.apply)
+
+# roughly based on http://davidwalsh.name/check-parent-node (note, OP is incorrect)
+contains = (parentNode, childNode) ->
+  if parentNode.contains
+    parentNode.contains(childNode)
+  else # old browser compatability
+    !!((parentNode == childNode) || (parentNode.compareDocumentPosition(childNode) & 16))
```

Comment:
We can change the 16 to `Node.DOCUMENT_POSITION_CONTAINED_BY`
https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition